app/components-react/root/GLVolmeters.tsx

```diff
@@ -0,0 +1,438 @@
+
+  // start rendering volmeters when the canvas is ready
+  useEffect(() => {
+    assertIsDefined(canvasRef.current);
+    setupNewCanvas(canvasRef.current);
+  },[]);
+
+  return (
+    <div style={{ position: 'absolute', height: '100%', width: '100%' }}>
+      <canvas
+        ref={canvasRef}
+        style={{
+          display: 'block',
+          position: 'absolute',
+          top: 0,
+          right: 0,
+          left: 0,
+          height: '100%',
+        }}
+      />
+    </div>
+  );
+}
+
+class GLVolmetersModule {
+  private customizationService = Services.CustomizationService;
+  private audioService = Services.AudioService;
+
+  subscriptions: Dictionary<IVolmeterSubscription> = {};
+
+  // Used for WebGL rendering
+  private gl: WebGLRenderingContext;
+  private program: WebGLProgram;
+
+  // GL Attribute locations
+  private positionLocation: number;
+
+  // GL Uniform locations
+  private resolutionLocation: WebGLUniformLocation;
+  private translationLocation: WebGLUniformLocation;
+  private scaleLocation: WebGLUniformLocation;
+  private volumeLocation: WebGLUniformLocation;
+  private peakHoldLocation: WebGLUniformLocation;
+  private bgMultiplierLocation: WebGLUniformLocation;
+
+  private canvasWidth: number;
+  private canvasWidthInterval: number;
+  private channelCount: number;
+  private canvasHeight: number;
+  private renderingInitialized: boolean;
+
+  // time between 2 received peaks.
+  // Used to render extra interpolated frames
+  interpolationTime = 35;
+  private bg: { r: number; g: number; b: number };
+  private fpsLimit: number;
+  private firstFrameTime: number;
+  private frameNumber: number;
+  private sourcesOrder: string[];
+  private workerId: number;
+  private requestedFrameId: number;
+  private bgMultiplier = this.customizationService.isDarkTheme ? 0.2 : 0.5;
+  private customizationServiceSubscription: Subscription = null!;
+
+  init() {
+    this.workerId = electron.ipcRenderer.sendSync('getWorkerWindowId');
+    this.subscribeVolmeters();
+    this.bg = this.customizationService.sectionBackground;
+    this.fpsLimit = this.customizationService.state.experimental!.volmetersFPSLimit! || 30;
+
+    // update FPS limit if settings have changed
+    this.customizationServiceSubscription = this.customizationService.settingsChanged.subscribe(
+      settings => {
+        if (settings.experimental?.volmetersFPSLimit) {
+          this.fpsLimit = settings.experimental.volmetersFPSLimit;
+        }
+      },
+    );
+  }
+
+  $refs = {
+    canvas: null! as HTMLCanvasElement,
+  };
+
+  // TODO: refactor into a single source of truth between Mixer and Volmeters
+  get audioSources() {
+    return this.audioService.views.sourcesForCurrentScene.filter(source => {
+      return !source.mixerHidden && source.isControlledViaObs;
+    });
+  }
+
+  /**
+   * add or remove subscription for volmeters depending on current scene
+   */
+  // @Watch('audioSources')
```

Comment:
is this meant to be commented out?

Reply:
Looks like a Vue `@Watch` decorator - most likely a copy paste from the Vue class.  Should remove.